tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -58,6 +58,37 @@ def _check_cpu_features_arm(test_microvm):
     )
 
 
+def get_cpu_template_dir(cpu_template):
+    """
+    Utility function to return a valid string which will be used as
+    name of the directory where snapshot artifacts are stored during
+    snapshot test and loaded from during restore test.
+
+    """
+    return cpu_template if cpu_template else "none"
+
+
+def skip_test_based_on_artifacts(snapshot_artifacts_dir):
+    """
+    It is possible that some X template is not supported on
+    the instance where the snapshots were created and,
+    snapshot is loaded on an instance where X is supported. This
+    results in error since restore doesn't find the file to load.
+    e.g. let's suppose snapshot is created on Skylake and restored
+    on Cascade Lake. So, the created artifacts could just be:
+    snapshot_artifacts/wrmsr/vmlinux-4.14/T2S
+    but the restore test would fail because the files in
+    snapshot_artifacts/wrmsr/vmlinux-4.14/T2CL won't be available.
+    To avoid this we make an assumption that if template directory
+    does not exist then snapshot was not created for that template
+    and we skip the test.
+    """
+    if not Path.exists(snapshot_artifacts_dir):
+        reason = f"\n Since {snapshot_artifacts_dir} does not exist \
+                we skip the test assuming that snapshot was not"
```

Comment:
maybe just `snapshot dir does not exist` to avoid the long string?

Reply:
I wanted to print the directory name to have more info in the logs.